src/transforms/eigenanalysis.jl

```diff
@@ -104,32 +112,36 @@ function reapply(transform::EigenAnalysis, table, cache)
     assertion(table)
   end
 
-  # original columns names
-  cols = Tables.columns(table)
-  names = Tables.columnnames(cols)
-
   # table as matrix
   X = Tables.matrix(table)
 
+  # output dimension
+  d = _ndim(transform.ndim, X)
+
   # retrieve cache
-  μ, S, S⁻¹ = cache
+  μ, S, S⁻¹, onames = cache
 
   # center the data
   Y = X .- μ
 
   # project the data
   Z = Y * S
 
+  # column names
+  names = Symbol.(:PC, 1:d)
+
   # table with transformed columns
   𝒯 = (; zip(names, eachcol(Z))...)
   𝒯 |> Tables.materializer(table)
 end
 
-function eigenmatrices(transform, Y)
+function eigenmatrices(transform, Y, d)
   proj = transform.proj
 
   Σ = cov(Y)
-  λ, V = eigen(Σ)
+  F = eigen(Σ)
+  λ = F.values[end:-1:1]
+  V = F.vectors[:, end:-1:1]
```

Comment:
Please use the sortby option of eigen instead and preserve the previous version with two output arguments.

Reply:
It is not possible to reverse the order using the sortby kwarg and the eigen function does not have `rev` kwarg like the sort function.

Reply:
Please read the docs of eigen, you can pass a sortby function. `eigen(A, sortby=l -> -real(l))` for example.